contrib/integration-cli-on-swarm/integration-cli-on-swarm.sh

```diff
@@ -0,0 +1,208 @@
+	docker stack rm $stack
+	# FIXME: make sure all resources are removed here
+        sleep 10
+	set +x
+    fi
+}
+
+cleanup_volume() {
+    [ $# -eq 0 ]
+    if docker inspect $volume > /dev/null 2>&1; then
+	log "Cleaning up volume $volume"
+	set -x
+	docker volume rm $volume
+	set +x
+    fi
+}
+
+ensure_images() {
+    [ $# -eq 1 ]; push_worker_image="$1"
+    log "Checking $master_image and $worker_image exists"
+    # We do not need to always build them. A user may want to run integration-cli-on-swarm.sh multiple times with the same image.
+    docker image inspect $master_image $worker_image > /dev/null || log "Please run \`make build-integration-cli-on-swarm\` first"
+    if [ $push_worker_image ]; then
+	log "Pushing $push_worker_image"
+	set -x
+	# TODO: skip pushing if no change since last push
+	docker image tag $worker_image $push_worker_image
+	docker image push $push_worker_image
+	set +x
+    fi
+}
+
+enum_filter_strings(){
+    [ $# -eq 0 ]
+    # TODO: refine the command for better maintainability.
+    #       Note that we could use `TESTFLAGS=-check.list make test-integration-cli`, but it is slow.
+    grep -oPh '^func \(.*\*\K\w+Suite\) Test\w+' integration-cli/*_test.go | sed -e 's/) /./g' | sed -e 's/$/\$/g' | sort
+    # The output will be as follows:
+    #  DockerAuthzSuite.TestAuthZPluginAPIDenyResponse$
+    #  DockerAuthzSuite.TestAuthZPluginAllowEventStream$
+    #  ...
+    #  DockerTrustedSwarmSuite.TestTrustedServiceUpdate$
+}
+
+create_volume() {
+    [ $# -eq 0 ]
+    log "Creating volume $volume"
+    set -x
+    docker volume create --driver local $volume
+    set +x
+}
+
+create_input() {
+    [ $# -le 1 ]; file="$1"
+    if [ -z $file ]; then
+	file=$(mktemp)
+	log "Generating the list of test filter strings as $file"
+	enum_filter_strings > $file
+    fi
+    log "Saving the list of test filter strings ($file) as a volume $volume"
+    set -x
+    cat $file | docker run -i --rm -v $volume:/mnt busybox sh -c "cat > /mnt/input"
+    set +x
+}
+
+create_compose_file() {
+    # TODO: use bash4 hash map (NOTE: macOS still ships with bash3)
+    [ $# -eq 6 ]; push_worker_image="$1" replicas="$2" chunks="$3" shuffle="$4" rand_seed="$5" dry_run="$6"
+    compose_worker_image=$worker_image
+    [ $push_worker_image ] && compose_worker_image=$push_worker_image
+    worker_image_id=$(docker inspect -f '{{.Id}}' $worker_image)
+    self_node_id=$(docker node inspect -f '{{.ID}}' self)
+    template=$(cat ./contrib/integration-cli-on-swarm/docker-compose.template.yml)
+    log "Creating ${compose_file}"
+    eval "echo \"${template}\"" > $compose_file
+}
+
+create_stack() {
+    [ $# -eq 0 ]
+    log "Creating stack $stack from ${compose_file}"
+    set -x
+    docker stack deploy --compose-file ${compose_file} --with-registry-auth $stack
+    set +x
+}
+
+inspect_master_container_id() {
+    [ $# -eq 0 ]
+    # FIXME(AkihiroSuda): we should not rely on internal service naming convention
+    docker ps --all --quiet \
```

Comment:
It is not sufficient to get the container ID (I need this ID to wait for its completion)